src/HeckeMiscMatrix.jl

```diff
@@ -0,0 +1,1225 @@
+    return diagonal_matrix(x)
+end
+
+################################################################################
+#
+#  Reduce the entries of a matrix modulo p
+#
+################################################################################
+
+@doc raw"""
+    mod!(M::ZZMatrix, p::ZZRingElem)
+
+Reduces every entry modulo $p$ in-place, i.e. applies the mod function to every entry.
+Positive residue system.
+"""
+function mod!(M::ZZMatrix, p::ZZRingElem)
+    GC.@preserve M begin
+        for i = 1:nrows(M)
+            for j = 1:ncols(M)
+                z = ccall((:fmpz_mat_entry, libflint), Ptr{ZZRingElem}, (Ref{ZZMatrix}, Int, Int), M, i - 1, j - 1)
+                ccall((:fmpz_mod, libflint), Nothing, (Ptr{ZZRingElem}, Ptr{ZZRingElem}, Ref{ZZRingElem}), z, z, p)
+            end
+        end
+    end
+    return nothing
+end
+
+@doc raw"""
+    mod(M::ZZMatrix, p::ZZRingElem) -> ZZMatrix
+
+Reduces every entry modulo $p$, i.e. applies the mod function to every entry.
+"""
+function mod(M::ZZMatrix, p::ZZRingElem)
+    N = deepcopy(M)
+    mod!(N, p)
+    return N
+end
+
+export mod_sym, mod_sym!
+
+@doc raw"""
+    mod_sym!(M::ZZMatrix, p::ZZRingElem)
+
+Reduces every entry modulo $p$ in-place, into the symmetric residue system.
+"""
+function mod_sym!(M::ZZMatrix, B::ZZRingElem)
+    @assert !iszero(B)
+    ccall((:fmpz_mat_scalar_smod, libflint), Nothing, (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZRingElem}), M, M, B)
+end
+mod_sym!(M::ZZMatrix, B::Integer) = mod_sym!(M, ZZRingElem(B))
+
+@doc raw"""
+    mod_sym(M::ZZMatrix, p::ZZRingElem) -> ZZMatrix
+
+Reduces every entry modulo $p$ into the symmetric residue system.
+"""
+function mod_sym(M::ZZMatrix, B::ZZRingElem)
+    N = zero_matrix(FlintZZ, nrows(M), ncols(M))
+    ccall((:fmpz_mat_scalar_smod, libflint), Nothing, (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZRingElem}), N, M, B)
+    return N
+end
+mod_sym(M::ZZMatrix, B::Integer) = mod_sym(M, ZZRingElem(B))
+
+
+@doc raw"""
+    mod_sym!(A::Generic.Mat{nf_elem}, m::ZZRingElem)
+
+Inplace: reduce all entries of $A$ modulo $m$, into the symmetric residue system.
+"""
+function mod_sym!(A::Generic.Mat{nf_elem}, m::ZZRingElem)
+    for i = 1:nrows(A)
+        for j = 1:ncols(A)
+            mod_sym!(A[i, j], m)
+        end
+    end
+end
+
+################################################################################
+#
+#  Special map entries
+#
+################################################################################
+
+function map_entries(R::zzModRing, M::ZZMatrix)
+    MR = zero_matrix(R, nrows(M), ncols(M))
+    ccall((:fmpz_mat_get_nmod_mat, libflint), Cvoid, (Ref{zzModMatrix}, Ref{ZZMatrix}), MR, M)
+    return MR
+end
+
+################################################################################
+#
+#  Concatenation of matrices
+#
+################################################################################
+
+@doc raw"""
+vcat(A::Vector{Generic.Mat}) -> Generic.Mat
+vcat(A::Array{ZZMatrix}, 1}) -> ZZMatrix
```

Comment:
Seems my scripts missed this one when we changed Array -> Vector/Matrix :-(
```suggestion
vcat(A::Vector{ZZMatrix}) -> ZZMatrix
```